schema.go

```diff
@@ -942,9 +708,21 @@ func mergeMapsOverride[T any](base, override map[string]T) map[string]T {
 	return base
 }
 
+func baseType(i any) reflect.Type {
```

Comment:
What is the benefit of this function? Does a `baseType(i)` have any properties that we care about?